components/scream/src/share/tests/field_tests.cpp

```diff
@@ -186,6 +189,16 @@ TEST_CASE("field_repo", "") {
   auto f5 = repo.get_field(fid5);
   auto f6 = repo.get_field(fid6);
 
+  // Check that get_field with a field name and grid name as arguments returns the appropriate field
+  // Using grid_1 should return fid2 field, using grid_2 should return fid3 field, using grid_3 should throw an error.
+  auto& f7 = repo.get_field("field_2","grid_1");
+  auto& f8 = repo.get_field("field_2","grid_2");
+  REQUIRE_THROWS( repo.get_field("field_2","grid_3") );
+  REQUIRE(f7.get_header().get_identifier()==fid2);
+  REQUIRE(f7.get_header().get_identifier()!=fid3);
+  REQUIRE(f8.get_header().get_identifier()!=fid2);
+  REQUIRE(f8.get_header().get_identifier()==fid3);
+
```

Comment:
We should also check that if there are two fields called `field_2` on `grid_2`, but with different layout, then `get_field(field_2,grid_2)` throws.

Reply:
Yes, good idea.  I'll put that in.